### PR TITLE
Add data consistency check pipeline stage

### PR DESCRIPTION
**Pull request by radix0000**
Add data consistency check pipeline stage, refactor and expand tests

https://github.com/openownership/bodsdata/issues/24

### CODE REVIEW

File: bodsdata.py [hunk 1]
```diff
@@ -995,18 +1023,25 @@ def run_pipeline(source, title, description, download, upload, bucket = ''):
         Upload to s3 bucket and delete local file, and update website
     bucket: string
         optional name of s3 bucket containing the source data
+    check: bool
+        optionally disable data consistency checks
+    check_is_component : bool
+        Optionally disable checking for isComponent in statements
```
Comment (odscjames) [inline bodsdata.py:1029]: Not all the new params are documented here, if it's clear elsewhere maybe that's ok

File: consistency_checks.py [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+                    examples = ", ".join(self.error_log)
+                else:
+                    examples = ", ".join(self.error_log[:5])
+                estats += f" ({examples})"
+                message = f"Consistency checks failed: {estats}"
+                raise AssertionError(message)
+
+    def run(self):
+        """Run consistency checks"""
+        self.read_data()
```
Comment (odscjames) [inline consistency_checks.py:196]: Not a requirement for this PR but in general if not all the above methods are meant to be run directly then marking them private with "_" would help readability.

File: bodsdata.py (hunk 1 above)
Comment (radix0000) [reply bodsdata.py:1031]: Done.

File: consistency_checks.py (hunk 2 above)
Comment (radix0000) [reply consistency_checks.py]: Done.

**Review by odscjames — APPROVED** [review]

**Pull request merged**